### PR TITLE
New CI fixes

### PR DESCRIPTION
**Pull request by tobias-urdin**
This contains new CI fixes.

* pyparsing >= 3.0.0 changed behavior of ``__bool__`` on pyparsing.ParseResult class
* Update asserRegex for gabbi tests where pyparsing changed the output
* Update mergify config to remove the ``strict`` key in the ``merge`` dict

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -4,7 +4,6 @@ pull_request_rules:
       merge:
         method: rebase
         rebase_fallback: merge
```
Comment (jd) [inline .mergify.yml:6]: might be better to switch to the queue action: https://blog.mergify.com/strict-mode-deprecation/

Comment (tobias-urdin) [reply .mergify.yml]: ack

**Comment by tobias-urdin** [comment]
Do not merge, pushed commit to test pyparsing change with pyparsing <3.0.0 versions.

File: .mergify.yml [hunk 2]
```diff
@@ -1,26 +1,46 @@
+queue_rules:
+  - name: default
+    conditions:
```
Comment (jd) [inline .mergify.yml:3]: if they are identical to condition to be put in queue you don't need to repeat them fwiw.

**Review by jd — APPROVED** [review]
I'd prefer 2 PR but ok :)

**Comment by jd** [comment]
I'm not clicking merge but you can (and need to) merge manually @tobias-urdin

**Comment by tobias-urdin** [comment]
@mergifyio backport stable/4.4

**Comment by tobias-urdin** [comment]
@mergifyio backport stable/4.3

**Pull request merged**